### PR TITLE
Emoji picker

### PR DESCRIPTION
**Pull request by CrisVega08**
## Summary 
- Add Picker for emojis
- Refactor Contenteditable element
- Caret helpers
- Solved Emojis #217 

#### Chrome
Chrome

#### Firefox
Firefox

#### Safari
Safari

### CODE REVIEW

File: src/utils/contentEditable.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  preSelectionRange.selectNodeContents(el)
+  preSelectionRange.setEnd(range.startContainer, range.startOffset);
+
+  const start = preSelectionRange.toString().length;
+  return { 
+    start: start,
+    end: start + range.toString().length
+  }
+}
+export const isFirefox = () =>  navigator.userAgent.search("Firefox") > 0;
```
Comment (mcallegari10) [inline src/utils/contentEditable.ts:26]:
```suggestion
export const isFirefox = () => navigator.userAgent.search("Firefox") > 0;
```

File: src/components/Widget/components/Conversation/components/Sender/index.tsx [hunk 2]
```diff
@@ -14,41 +17,123 @@ type Props = {
...
+
+  const handlerPressEmoji = () => {
+    onPressEmoji();
+    checkSize();
   }
 
   return (
-    <div className="rcw-sender">
+    <div ref={refContainer} className="rcw-sender">
+      <button className='rcw-picker-btn' type="submit" onClick={handlerPressEmoji}>
```
Comment (mcallegari10) [inline src/components/Widget/components/Conversation/components/Sender/index.tsx:129]: I think we should add a conditional to hide the emoji picker setting it to true by default. What do you think?

File: src/components/Widget/components/Conversation/components/Sender/index.tsx [hunk 3]
```diff
@@ -14,41 +17,123 @@ type Props = {
...
+  }
+
+  const handlerOnKeyUp = (event) => {
+    const el = inputRef.current;
+    if(!el) return true;
+    // Conditions need for firefox
+    if(firefox && event.key === 'Backspace') {
+      if(el.innerHTML.length === 1 && enter) {
+        el.innerHTML = ''
+        setEnter(false)
```
Comment (mcallegari10) [inline src/components/Widget/components/Conversation/components/Sender/index.tsx:98]:
```suggestion
        el.innerHTML = '';
        setEnter(false);
```

File: src/components/Widget/components/Conversation/components/Sender/index.tsx [hunk 4]
```diff
@@ -14,41 +17,123 @@ type Props = {
...
+      }
+      else if(brRegex.test(el.innerHTML)){
+        el.innerHTML = el.innerHTML.replace(brRegex, '')
```
Comment (mcallegari10) [inline src/components/Widget/components/Conversation/components/Sender/index.tsx:101]:
```suggestion
        el.innerHTML = el.innerHTML.replace(brRegex, '');
```

File: src/components/Widget/components/Conversation/components/Sender/index.tsx [hunk 5]
```diff
@@ -14,41 +17,123 @@ type Props = {
...
+      }
+    }
+    checkSize()
```
Comment (mcallegari10) [inline src/components/Widget/components/Conversation/components/Sender/index.tsx:104]:
```suggestion
    checkSize();
```

File: src/scss/_common.scss [hunk 6]
```diff
@@ -4,10 +4,6 @@
   max-width: 215px;
   padding: 15px;
   text-align: left;
-  
-  > p {
-    white-space: pre-wrap;
-  }
```
Comment (mcallegari10) [inline src/scss/_common.scss:10]: Is it ok to delete this? Won't be a problem when using markdown?

File: src/utils/contentEditable.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  selection.removeAllRanges();
+  selection.addRange(range);
+  el.focus();
+}
+
+export const insertNodeAtCaret = (el) => {
+  const position = getCaretIndex(el)
+  let characterToEnter = '\n\n';
+  let prevChar, char = '';
+  if (position > 0) { // Change this
```
Comment (mcallegari10) [inline src/utils/contentEditable.ts:42]: Change this?

**Review by mcallegari10 — CHANGES_REQUESTED** [review]
Remember to update the readme with the new feature 😄

File: src/scss/_common.scss (hunk 6 above)
Comment (CrisVega08) [reply src/scss/_common.scss:10]: Yes its ok, this style I added in before PR.

**Review by mcallegari10 — APPROVED** [review]

**Comment by Bonfims** [comment]
Hi, awesome job with react-chat-widget.

Is a question about emoji feature, can we set their properties like is designed on [emoji-mart git page](https://github.com/missive/emoji-mart)?

If yes, how can i do that? Would be nice to set [i18n](https://github.com/missive/emoji-mart#i18n) propertie to do some translation on it.

Thanks

note: maybe i should have posted this in the questions tab, sorry

**Pull request merged**